tests/functional/commands/test_db_load.py

```diff
@@ -0,0 +1,462 @@
+#!/usr/bin/env python
+"""Perform functional tests for db load."""
+
+import os
+
+import orion.core.cli
+from orion.storage.base import setup_storage
+
+LOAD_DATA = os.path.join(os.path.dirname(__file__), "orion_db_load_test_data.pickled")
+LOAD_DATA_WITH_BENCHMARKS = os.path.join(
+    os.path.dirname(__file__), "orion_db_load_test_data_with_benchmarks.pickled"
+)
+
+
+# TODO: Test loading benchmarks
+
+
+def execute(command, assert_code=0):
+    """Execute orion command and return returncode"""
+    returncode = orion.core.cli.main(command.split(" "))
+    assert returncode == assert_code
+
+
+def common_indices(data_list1, data_list2):
+    """Return set of common indices from two lists of data"""
+    return {element["_id"] for element in data_list1} & {
+        element["_id"] for element in data_list2
+    }
+
+
+def test_empty_database(empty_database):
+    """Test destination database is empty as expected"""
+    storage = setup_storage()
+    db = storage._db
+    with db.locked_database(write=False) as internal_db:
+        collections = set(internal_db._db.keys())
+    assert collections == {"experiments", "algo", "trials", "benchmarks"}
+    assert len(db.read("experiments")) == 0
+    assert len(db.read("algo")) == 0
+    assert len(db.read("trials")) == 0
+    assert len(db.read("benchmarks")) == 0
+
+
+def test_load_all(empty_database):
+    """Test load all database"""
+    assert os.path.isfile(LOAD_DATA_WITH_BENCHMARKS)
+    storage = setup_storage()
+    loaded_db = storage._db
+    assert len(loaded_db.read("benchmarks")) == 0
+    assert len(loaded_db.read("experiments")) == 0
+    assert len(loaded_db.read("trials")) == 0
+    assert len(loaded_db.read("algo")) == 0
+
+    execute(f"db load {LOAD_DATA_WITH_BENCHMARKS}")
+    with loaded_db.locked_database(write=False) as internal_db:
+        collections = set(internal_db._db.keys())
+    assert collections == {"experiments", "algo", "trials", "benchmarks"}
+
+    assert len(loaded_db.read("benchmarks")) == 3
+    assert len(loaded_db.read("experiments")) == 3
+    assert len(loaded_db.read("trials")) == 24
+    # TODO: We should expect 6 algorithms, but only 3 are returned
```

Comment:
I don't understand. The `pickle` file contains 6 algo locks, but only 3 experiments? This should not be possible. How was the file generated?

Reply:
It's likely because we update the experiment `_id` here: https://github.com/notoraptor/orion/blob/import-export-cli-and-webapi/tests/functional/commands/conftest.py#L51. And we do not fix this id in the algo lock. It would need to be fixed.

Reply:
Done: https://github.com/Epistimio/orion/pull/1082/commits/ebfece3f477c509bab501736937b1e8e816f9a72